File: Introduction to Data Structures: Hash Table/5. Conclusion/1. Jewels & Stones/Jewels&Stones.java
class Solution {
    public int numJewelsInStones(String jewels, String stones) {
        
        Map<Character, Integer> hashMap = new HashMap<>();
        
        int total = 0;
        
        for (int i = 0; i < jewels.length(); i++) {
            hashMap.put(jewels.charAt(i), 0);
        }
        
        for (int i = 0; i < stones.length(); i++) {
            if (hashMap.containsKey(stones.charAt(i))) {
                total += 1;
            }
        }
        
        return total;
    }
}

File: Introduction to Data Structures: Hash Table/5. Conclusion/1. Jewels & Stones/Jewels&Stones.py
class Solution:
    def numJewelsInStones(self, jewels: str, stones: str) -> int:

        hashMap = {}

        for j in jewels:

            hashMap[j] = 0

        for s in stones:

            if hashMap.get(s) != None:

                hashMap[s] += 1

        return sum(hashMap.values())
